core/src/main/java/edu/ucr/cs/riple/core/Annotator.java

```diff
@@ -148,7 +155,27 @@ private void explore() {
               .filter(report -> report.getOverallEffect(config) < 1)
               .flatMap(report -> config.chain ? report.tree.stream() : Stream.of(report.root))
               .collect(Collectors.toSet()));
-      if (sizeBefore == this.reports.size()) {
+      // Collect impacted parameters from changes in target module due to usages in downstream
+      // dependencies.
+      if (config.downStreamDependenciesAnalysisActivated) {
+        triggeredFixesFromDownstreamDependencies =
+            latestReports.stream()
+                .flatMap(
+                    report ->
+                        downStreamDependencyExplorer.getImpactedParameters(report.tree).stream()
```

Comment:
If I fully understood the question, what happens here is that we only query once what are the side effects of `getNull()` and if it is considered as a useful fix, it will be injected (followed by process of impacted parameters) or otherwise will be tagged as unuseful permanently. Therefore, we do not need to update the state after the injection. Please let me know if that does not cover the case you mentioned. 🙂

Reply:
We discussed this offline and should be addressed in a follow up PR, including the test cases above.

Reply:
The requested feature is described here #62.